src/Arcus.Observability.Tests.Unit/Telemetry/TelemetryTypeFilterTests.cs

```diff
@@ -0,0 +1,114 @@
+        {
+            // Arrange
+            DateTimeOffset timestamp = _bogusGenerator.Date.RecentOffset();
+            var level = _bogusGenerator.Random.Enum<LogEventLevel>();
+            var telemetryType = _bogusGenerator.Random.Enum<TelemetryType>();
+            var properties = new[]
+            {
+                new LogEventProperty(ContextProperties.General.TelemetryType, new ScalarValue(telemetryType))
+            };
+
+            var logEvent = new LogEvent(timestamp, level, exception: null, MessageTemplate.Empty, properties);
+            var filter = TelemetryTypeFilter.On(telemetryType);
+            
+            // Act
+            bool isEnabled = filter.IsEnabled(logEvent);
+            
+            // Assert
+            Assert.False(isEnabled);
+        }
+
+        [Fact]
+        public void LogEventWithoutTelemetry_DoesntFilterAnything_Succeeds()
+        {
+            // Arrange
+            DateTimeOffset timestamp = _bogusGenerator.Date.RecentOffset();
+            var level = _bogusGenerator.Random.Enum<LogEventLevel>();
+            var telemetryType = _bogusGenerator.Random.Enum<TelemetryType>();
+            var logEvent = new LogEvent(timestamp, level, exception: null, MessageTemplate.Empty, Enumerable.Empty<LogEventProperty>());
+            var filter = TelemetryTypeFilter.On(telemetryType);
+            
+            // Act
+            bool isEnabled = filter.IsEnabled(logEvent);
+            
+            // Assert
+            Assert.False(isEnabled);
+        }
+
+        [Fact]
+        public void LogEventWithoutValidTelemetry_DoesntFilterAnything_Succeeds()
+        {
+            // Assert
+            DateTimeOffset timestamp = _bogusGenerator.Date.RecentOffset();
+            var level = _bogusGenerator.Random.Enum<LogEventLevel>();
+            var telemetryType = _bogusGenerator.Random.Enum<TelemetryType>();
+            var logEvent = new LogEvent(timestamp, level, exception: null, MessageTemplate.Empty, Enumerable.Empty<LogEventProperty>());
+            var filter = TelemetryTypeFilter.On(telemetryType);
+            
+            // Act
+            bool isEnabled = filter.IsEnabled(logEvent);
+            
+            // Assert
+            Assert.False(isEnabled);
+        }
+
+        [Theory]
+        [InlineData(TelemetryType.Request | TelemetryType.Dependency)]
+        [InlineData(TelemetryType.Metrics | TelemetryType.Request | TelemetryType.Events)]
+        public void CreateFilter_WithTelemetryTypeOutsideEnum_Fails(TelemetryType telemetryType)
+        {
+            Assert.ThrowsAny<ArgumentException>(() => TelemetryTypeFilter.On(telemetryType));
+        }
+        
+        [Theory]
+        [InlineData(TelemetryType.Request | TelemetryType.Dependency)]
+        [InlineData(TelemetryType.Metrics | TelemetryType.Request | TelemetryType.Events)]
+        public void CreateFilterWithTracked_WithTelemetryTypeOutsideEnum_Fails(TelemetryType telemetryType)
```

Comment:
I'm not sure if I get the test case here, can you elaborate please?

Reply:
The telemetry filter guards against misused telemetry type enumeration values. All the inputs must be within the bounds of the enumeration.
This test checks that values outside the scope of the enumeration will result in argument exceptions.